### DISCUSSION TITLE
Many-to-one relations in grid

### DISCUSSION THREAD

**Discussion by ilfaro** [pimcore/pimcore#14192] [Q&A / Support]
I'm defining some custom grids for Data Objects.

In order to display the "name" attribute for my many-to-one relation fields,  I'm using an Operator ObjectField Getter.
Doing so, however, disables both filtering and sorting functionalities on the column.

Is there a way to display a user friendly name (instead of the full path) for the related object while retaining all sorting and filtering functionalities?

Thank you in advance

**Comment by SamyMP** [comment]
Hi,

Unfortunately, there is no way to do that at the moment, at least not that I know of. I would also be interested in having such a feature but yeah, using any grid operator would disable filtering and sorting (even editing) of the column 😕

**Reply by ilfaro** [reply]
Thank you for your reply Samy.

Do you happen to know a different workaround to make those grids more user friendly?

Clients often really don't like those ugly full paths

**Reply by SamyMP** [reply]
I don't know if that can suit your cases, but there are 2 ways to make them more user friendly, both requiring a bit of code :
1. Use a [Calculated Value](https://pimcore.com/docs/pimcore/current/Development_Documentation/Objects/Object_Classes/Data_Types/Calculated_Value_Type.html) which displays only the name of the attribute instead of its path. This requires you modify your model to add this field. You can filter on it since it is considered as "text" input, but you can't filter by relation for example

2. Use a [javascript event listener](https://github.com/pimcore/pimcore/blob/11.x/bundles/AdminBundle/public/js/pimcore/events.js). When the grid is displayed, use javascript to detect all paths and display only the part you want. This method assumes the name is in the path. The advantage of this method is that you'll still be able to filter by relation

I don't know if you're familiar with PHP/Javascript, but that's currently the only ways you can make the grid better. Hope this helps !

**Reply by ilfaro** [reply]
Thank you again for your suggestions.

I think I'll go for the javascript solution.

Which event do you think I should listen to? I see there's a "preCreateObjectGrid" event but not a "postCreateObjectGrid", which I think would be the most logical choice.

Sorry to keep pestering you but I'm very much new to Pimcore

**Reply by SamyMP** [reply]
No problem, don't worry, we've all been there 😉

The "preCreateObjectGrid" doesn't have the list of objects unfortunately, so you won't be able to change anything at this level. I would need to dig a little bit more to give you a better answer but for now, you could look at [Mutation Observer](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver). It allows you to interact with dom elements as they load, so when you see something that looks like a path, you could edit the textContent attribute of the concerned node to edit the cell content. 

Again, not the best solution but that's the "quick and dirty" answer I can give you for now. I'll investigate when I have time and come back to you with a better answer if I find one :)
